tests/podsmiter/src/main.rs

```diff
@@ -0,0 +1,185 @@
+use k8s_openapi::api::core::v1::{Namespace, Pod};
+use k8s_openapi::apimachinery::pkg::apis::meta::v1::ObjectMeta;
+use k8s_openapi::Metadata;
+use kube::api::{Api, DeleteParams, ListParams};
+
+#[tokio::main(threaded_scheduler)]
+async fn main() -> anyhow::Result<()> {
+    let result = smite_all_integration_test_pods().await;
+
+    match &result {
+        Ok(message) => println!("{}", message),
+        Err(e) => println!("{}", e),
+    };
+
+    result.map(|_| ())
+}
+
+async fn smite_all_integration_test_pods() -> anyhow::Result<&'static str> {
+    let client = match kube::Client::try_default().await {
+        Ok(c) => c,
+        Err(e) => {
+            return Err(anyhow::anyhow!(
+                "Failed to acquire Kubernetes client: {}",
+                e
+            ))
+        }
+    };
+
+    let namespaces = list_e2e_namespaces(client.clone()).await?;
+
+    if namespaces.is_empty() {
+        return Ok("No e2e namespaces found");
+    }
+    if !confirm_smite(&namespaces) {
+        return Ok("Operation cancelled");
+    }
+
+    let pod_smite_operations = namespaces
+        .iter()
+        .map(|ns| smite_namespace_pods(client.clone(), ns));
+    let pod_smite_results = futures::future::join_all(pod_smite_operations).await;
+    let (_, pod_smite_errors) = pod_smite_results.partition_success();
+
+    if !pod_smite_errors.is_empty() {
+        return Err(anyhow::anyhow!(smite_failure_message(&pod_smite_errors)));
+    }
+
+    println!("Requested force-delete of all pods; requesting delete of namespaces...");
+
+    let ns_smite_operations = namespaces
+        .iter()
+        .map(|ns| smite_namespace(client.clone(), ns));
+    let ns_smite_results = futures::future::join_all(ns_smite_operations).await;
+    let (_, ns_smite_errors) = ns_smite_results.partition_success();
+
+    if !ns_smite_errors.is_empty() {
+        return Err(anyhow::anyhow!(smite_failure_message(&pod_smite_errors)));
+    }
+
+    Ok("All e2e pods force-deleted; namespace cleanup may take a couple of minutes")
+}
+
+async fn list_e2e_namespaces(client: kube::Client) -> anyhow::Result<Vec<String>> {
+    println!("Finding e2e namespaces...");
+
+    let nsapi: Api<Namespace> = Api::all(client.clone());
+    let nslist = nsapi.list(&ListParams::default()).await?;
+
+    Ok(nslist
+        .iter()
+        .map(name_of)
+        .filter(|n| is_e2e_namespace(n))
```

Comment:
Nit: I think this can just be shortened to `filter(is_e2e_namespace)`. Clippy might even complain about it as is right now

Reply:
I thought that but apparently not; if I just pass the function directly then I get a type mismatch error.  The lambda form seems to unify `&str` and `&String` in a way that the direct form doesn't.  It's weird, and I agree that what you describe _should_ work... but...